File: Codes/0693-binary-number-with-alternating-bits_20260218.go
/*
 * @Author: LetMeFly
 * @Date: 2026-02-18 14:51:23
 * @LastEditors: LetMeFly.xyz
 * @LastEditTime: 2026-02-18 15:01:13
 */
package main

func hasAlternatingBits(n int) bool {
    x := (n >> 1) ^ n
    return (x & (x + 1)) == 0
}

File: Codes/0693-binary-number-with-alternating-bits_20260218.rs
/*
 * @Author: LetMeFly
 * @Date: 2026-02-18 14:51:23
 * @LastEditors: LetMeFly.xyz
 * @LastEditTime: 2026-02-18 14:59:11
 */
impl Solution {
    pub fn has_alternating_bits(n: i32) -> bool {
        let x: usize = ((n >> 1) ^ n )as usize;
        (x & (x + 1)) == 0
    }
}